src/api.coffee

```diff
@@ -47,15 +47,41 @@ module.exports = (application) ->
 		application.update(req.body.force)
 		res.sendStatus(204)
 
-	unparsedRouter.post '/v1/reboot', (req, res) ->
+	parsedRouter.post '/v1/reboot', (req, res) ->
+		force = req.body.force
 		utils.mixpanelTrack('Reboot')
-		request.post(config.gosuperAddress + '/v1/reboot')
-		.pipe(res)
+		knex('app').select()
+		.then (apps) ->
+			Promise.map apps, (theApp) ->
+				Promise.using application.lockUpdates(theApp.appId, force), ->
+					# There's a slight chance the app changed after the previous select
+					# So we fetch it again now the lock is acquired
+					knex('app').select().where({ appId: theApp.appId })
+					.then ([ app ]) ->
+						application.kill(app, true, false) if app?
```

Comment:
It'd be really nice to convert `application.kill` to use an options object instead of these booleans, because I have no idea what they denote from this context